### DISCUSSION TITLE
Observed discrepancy in Xonsh alias behavior between Linux and Windows

### DISCUSSION THREAD

**Discussion by tsobczynski** [xonsh/xonsh#4824] [Q&A]
I am observing an issue trying to alias 'ls' to my liking across Xonsh shells on both Linux and Windows. Is this just operator error on my part, or could there be a genuine issue here with Xonsh on Windows?

On Linux, I can use the alias for 'ls' as I would expect:

```
>>> 'ls' in aliases
True
>>> aliases['ls']
['ls', '--color=auto', '-v']
>>> ls --group-directories-first -lgG xonsh | head -4
total 296
drwxr-xr-x 1   4096 May 25 13:11 appimage
drwxr-xr-x 1   4096 May 25 13:11 ci
drwxr-xr-x 1   4096 May 25 16:21 docs

```

But on Windows, where I pick up 'ls' from Git for Windows (i.e. a subset of MSYS2) and it works normally, I can't use 'ls' at all once it has been aliased. For simplicity of a reproducer, I'm trying to use the same alias that is the default for Xonsh on Linux with the same sample arguments:

```
>>> 'ls' in aliases
False
>>> which ls
C:\Program Files\Git\usr\bin\ls.exe

>>> ls --group-directories-first -lgG xonsh | head -4
total 375
drwxr-xr-x 1      0 May 25 13:11 appimage
drwxr-xr-x 1      0 May 25 13:11 ci
drwxr-xr-x 1      0 May 25 16:21 docs
>>> aliases['ls'] = ['ls', '--color=auto', '-v']
>>> ls --group-directories-first -lgG xonsh | head -4
xonsh: subprocess mode: permission denied: ls

>>> aliases['ls']
['ls', '--color=auto', '-v']
>>> ls xonsh
xonsh: subprocess mode: permission denied: ls
```

**Comment by jnoortheen** [comment]
It is a known error -https://github.com/xonsh/xonsh/issues/4566. Currently recursive alias names aren't handled on windows. You can use a different name for the alias.
